### PR TITLE
Adds error handling messages for the user when they should be adding …

### PR DESCRIPTION
**Pull request by vimota**
…a BQ integration

### CODE REVIEW

**Comment by vimota** [comment]
This is what it will look like for the "public bq trying to access private dataset' case . I chose to use print() instead of modifiying the error message because it is more visible this way, the stack trace covers up the actual message string and you have to scroll to see it.
image

File: patches/kaggle_gcp.py [hunk 1]
```diff
@@ -52,6 +56,16 @@ def __init__(self, client):
         self._EXTRA_HEADERS["X-KAGGLE-PROXY-DATA"] = os.getenv(
             "KAGGLE_DATA_PROXY_TOKEN")
 
+    def api_request(self, *args, **kwargs):
+        """Wrap Connection.api_request in order to handle errors gracefully.
+        """
+        try:
+            super().api_request(*args, **kwargs)
+        except Forbidden as e:
+            print("Got a Permission Denied using Kaggle's Public Tier. "
```
Comment (harrisse) [inline patches/kaggle_gcp.py]: Proposed text: "Permission denied using Kaggle's public BigQuery integration. Did you mean to select a BigQuery account in the Kernels Settings sidebar?"

File: patches/sitecustomize.py [hunk 2]
```diff
@@ -15,14 +15,17 @@
...
+        has_bigquery = get_integrations().has_bigquery()
+        if specified_project is None and specified_credentials is None and not has_bigquery:
             print("Using Kaggle's public dataset BigQuery integration.")
             return PublicBigqueryClient(*args, **kwargs)
 
         else:
             if specified_credentials is None:
                 kwargs['credentials'] = KaggleKernelCredentials()
+                if (not has_bigquery):
+                    print('Please ensure you have enabled a BigQuery '
```
Comment (harrisse) [inline patches/sitecustomize.py]: "ensure you have selected"

**Comment by vimota** [comment]
Was that merge by accident? :)

**Pull request merged**